test/acceptance/tests/hcp/hcp_test.go

```diff
@@ -155,6 +150,402 @@ func TestHCP(t *testing.T) {
 	logger.Log(t, "Test successful!")
 }
 
-type listTasksResponse struct {
-	TaskARNs []string `json:"taskArns"`
+// TestNamespaces ensures that services in different namespaces can be
+// can be configured to communicate.
+func TestNamespaces(t *testing.T) {
+	// read the configuration from the setup-terraform dir.
+	var cfg HCPTestConfig
+	require.NoError(t, config.UnmarshalTF(setupDir, &cfg))
+
+	// generate input variables to the test terraform using the config.
+	ignoreVars := []string{"ecs_cluster_1_arn", "ecs_cluster_2_arn"}
+	tfVars := config.TFVars(cfg, ignoreVars...)
+
+	consulClient, consulState, err := consulClient(t, cfg.ConsulAddr, cfg.ConsulToken)
```

Comment:
I think naming `consulState` something like `initialConsulState` would be more clear about its intention. This threw me off for a while.

Reply:
Agreed.. Sorry for the confusion.